### DISCUSSION TITLE
`derive` `serde::{Serialize, Deserialize}` for `Error`

### DISCUSSION THREAD

**Discussion by bicarlsen** [Byron/trash-rs#118] [Q&A]
Could we `derive` `serde::{Serialize, Deserialize}` for `Error`? Perhaps placing it behind a feature flag.

**Comment by bicarlsen** [comment]
Some of the fields aren't `Serialize` or `Deserialize` so a straight `derive` won't work, unless they are skipped, or we implement them manually. I'm happy to do a PR for this if it's something you would want.

**Comment by Byron** [comment]
Somehow I feel that it would go beyond the scope of this crate. For instance, an `std::io::Error` also doesn't implement `serde`, so one would have to do that here in a way that won't be canonical.
Maybe I am missing something, but it seems like it's easier to convert the error into an own type that implements serde natively, and do so downstream.
Does that make sense?

**Reply by bicarlsen** [reply]
Yeah. After I dug in a bit, I agree with you. It seems we would have to take too many decisions on how to implement things for it to make sense to do it here.
